app/_components/TopicSelector.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+
+import clsx from 'clsx';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+
+import { DeveloperType } from '@/app/_types/interview';
+
+import SelectButton from './SelectButton';
+import styles from './TopicSelector.module.scss';
+
+interface TopicSelectorProps {
+  variant?: 'topic' | 'subTopic';
+  devType: DeveloperType;
+  topics: string[];
+}
+
+export default function TopicSelector({ variant = 'topic', devType, topics }: TopicSelectorProps) {
+  const router = useRouter();
+  const [selectedTopics, setSelectedTopics] = useState<string[]>([]);
+
+  const handleNext = () => {
+    if (!selectedTopics.length) return;
+
+    switch (variant) {
+      case 'topic':
+        const topicParam = selectedTopics.length === topics.length ? `all` : selectedTopics.join(',');
+        router.push(`/interview/select/${devType}/prepare?topics=${topicParam}`);
+        break;
+      case 'subTopic':
+        router.push(`/interview/chat`);
+        break;
+    }
+  };
+
+  const handleClickTopic = (clickedTopic: string) => {
+    setSelectedTopics((prev) =>
+      prev.includes(clickedTopic) ? prev.filter((topic) => topic !== clickedTopic) : [...prev, clickedTopic]
+    );
+  };
```

Comment:
```suggestion
const handleClickTopic = (clickedTopic: string) => {
    if (selectedTopics.includes(clickedTopic)) {
      const newSelectedTopics = selectedTopics.filter((topic) => topic !== clickedTopic);
      setSelectedTopics(newSelectedTopics);
      return;
    }

    setSelectedTopics([...selectedTopics, clickedTopic]);
  };
```

삼항 연산자로 처리하는 부분이 코드를 한눈에 보기 어려운 것 같아요

Reply:
저 궁금한 점 있어요!!
1. 함수형 업데이트를 권장하므로 저렇게 사용하는게 좋지않은지?
2. 슈니가 수정해준식으로 고친다면 handleClickTopic에 selectedTopics을 인자로 전달하는 건 어떻게 생각하는지?!(함수형프로그래밍관점..?)

Reply:
1. 함수형 프로그래밍 관점보다 가독성 측면에서 해당 리뷰 드렸어요. 모든 로직이 콜백 함수 안에 숏코딩 형식으로 있다 보니까 코드를 바로바로 이해하기 어려웠었던 것 같아요
2. 인자를 배열로 전달하신다는 말씀이신지 아니면 selectedTopics를 추가로 전달한다는 말씀이신지는 모르겠는데, clickedTopic만 전달하는 것은 그렇게 어색하지는 않은 것 같아요